rtengine/image16.cc

```diff
@@ -284,6 +284,46 @@ void Image16::getStdImage (ColorTemp ctemp, int tran, Imagefloat* image, Preview
 #undef GCLIP
 }
 
+bool
+Image16::isBW() const
+{
+    if (!height || !width) {
+        return false;
+    }
+
+    bool res = true;
+
+    for ( int w = 0, h = height / 2; res && w < width; ++w ) {
+        res =
+            res
+            && (
+                r(h, w) == g(h, w)
+                || r(h, w) == b(h, w)
```

Comment:
@adamreichold My idea was to include the early return check in the inner loop of parallel region. That should break the loop quite early for non b&w images and avoids extra code for checking the middle line. Of course it could happen that one core gets an area with only b&w pixels and but that's really unlikely for non b&w images.